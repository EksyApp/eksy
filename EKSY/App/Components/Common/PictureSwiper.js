import React, {Component} from 'react'
import Swiper from 'react-native-swipe-a-lot'
import {View, StyleSheet} from "react-native";
import Image from "./Image";
import FastImage from 'react-native-fast-image'

export default class PictureSwiper extends Component {
	
	render() {
		return (
				<Swiper wrapperStyle={[styles.swiper, this.props.style]}>
					{this.props.data.map((image, index) => {
						return (
								<View key={image.uri + index}>
									<Image
											style={styles.image}
											source={{uri: image.uri}}
											resizeMode={FastImage.resizeMode.contain}
									/>
								</View>
						)
					})}
				</Swiper>
		)
	}
	
}

const styles = StyleSheet.create({
	swiper: {
		height: '100%',
		width: '100%'
	},
	
	image: {
		width: '100%',
		height: '100%'
	}
})