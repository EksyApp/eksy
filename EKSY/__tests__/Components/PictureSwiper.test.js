import React from 'react';
import PictureSwiper from '../../App/Components/Common/PictureSwiper';
import renderer from 'react-test-renderer';

test('PictureSwiper renders correctly', () => {
	const tree = renderer.create(
			<PictureSwiper data = {[{width: 320, height: 240, uri: 'http://www.catpic.com'}]}/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
