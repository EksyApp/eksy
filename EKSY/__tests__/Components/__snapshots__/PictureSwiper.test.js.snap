// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureSwiper renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ],
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    bounces={false}
    horizontal={true}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <View>
          <FastImageView
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            resizeMode="contain"
            source={
              Object {
                "uri": "http://www.catpic.com",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "height": 32,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(158, 158, 158)",
              "borderRadius": 6,
              "height": 12,
              "margin": 10,
              "width": 12,
            },
            undefined,
            Object {
              "backgroundColor": "rgb(245, 245, 245)",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
